src/qt/optionsdialog.cpp

```diff
@@ -64,6 +69,7 @@ void OptionsDialog::setEnableState()
     this->ui->pushButton_start->setEnabled(mb_started);
     this->ui->spin_domainId->setEnabled(mb_started);
     this->ui->pushButton_stop->setEnabled(!mb_started);
+    this->ui->statisticsCheckBox->setEnabled(mb_started);
```

Comment:
What happens with new statistics feature if shapes-demos was compiled against a Fast-DDS without statistics?
Could we include `#include <fastdds/config.h>` and use the sample define to enable the feature at compilation time in the shapes-demo?

Reply:
Nothing as the design is robust and will not break.
However, an additional `#def` has been added to not enable the checkbox in case the Statistics are not compiled.